src/v/cloud_roles/apply_credentials.cc

```diff
@@ -10,6 +10,19 @@
 
 #include "cloud_roles/apply_credentials.h"
 
+#include "cloud_roles/apply_aws_credentials.h"
+
+cloud_roles::apply_credentials
+cloud_roles::make_credentials_applier(cloud_roles::credentials creds) {
+    return apply_credentials{ss::visit(
+      std::move(creds),
+      [](cloud_roles::aws_credentials ac)
+        -> std::unique_ptr<apply_credentials::impl> {
```

Comment:
in this instance it seems to be required because `ss::visit` breaks without it:

```
In template: static_assert failed due to requirement '__visit_rettypes_match' "std::visit requires the visitor to have the same return type for all alternatives of a variant"
```

I had no trailing return originally but `ss::visit` complained of differing types.